software/include/enso/consts.h

```diff
@@ -135,19 +135,31 @@ static constexpr std::string_view kIpcQueueToAppName = "enso_ipc_queue_to_app";
 static constexpr std::string_view kIpcQueueFromAppName =
     "enso_ipc_queue_from_app";
 
-enum class MmioNotifType : uint64_t {
+enum class NotifType : uint8_t {
   kWrite = 0,
   kRead = 1,
-  kTranslAddr = 2  // Used to translate physical address to virtual address in
-                   // the software backend address space.
+  kTranslAddr = 2,  // Used to translate physical address to virtual address in
+                    // the software backend address space.
+  kAllocatePipe = 3,
+  kAllocateNotifBuf = 4,
+  kGetNbFallbackQueues = 5,
+  kSetRrStatus = 6,
+  kGetRrStatus = 7,
+  kFreeNotifBuf = 8,
+  kFreePipe = 9
 };
 
 struct MmioNotification {
-  MmioNotifType type;
+  NotifType type;
   uint64_t address;
   uint64_t value;
 };
 
+struct PipeNotification {
+  NotifType type;
+  uint64_t data[6];
```

Comment:
Can you be more specific here? Instead of using `data[6]`, can you name the fields? This makes it much easier to read the code.

Reply:
The thing for this is each notification struct would have different fields: do you think I should create a different struct for each? Each of them would only require one element of the data array, so the structure would just be (for example)
`struct SetRrStatusNotification {
    NotifType type;
    uint64_t rr_status;
}`

would you prefer that? There would just be like 10 structures with a similar format.

Reply:
Would you mind doing that? It's just that as it is it's actually less readable than it used to be.